### ISSUE TITLE
Improvement of internal structure: Wrapping Git commands

### ISSUE THREAD

**Issue by kitsuyui** [kitsuyui/mure#88]
Currently, Git commands are wrapped in `src/git.rs`.
This implementation has the following problems.

- Wrap the results of raw Git commands, so it is harder for users to understand what is happening than to run Git commands directly
- There are too many combinations of wrapped processes, and it is difficult to implement all of them
- However, I want to safely handle the known process results with Rust's enum

## Improvement plan

Define a result struct. It will be structured as follows

1. Struct that represents the result of the raw Git command you are looking for
  - Exit status code
  - Standard output
  - Standard error output
2. Enum, struct that represents the complete Rust processing result you are looking for (this will change depending on the expression you want to represent)
3. A generic struct that can hold 1 and 2

If this approach is adopted, it will also be a great help when implementing the quiet, verbose options that described in
 https://github.com/kitsuyui/mure/issues/11 .
In other words, if verbose is specified, just output the message contained in the struct of 1.
